internal/services/applicationinsights/application_insights_resource.go

```diff
@@ -228,7 +228,11 @@ func resourceApplicationInsightsCreateUpdate(d *pluginsdk.ResourceData, meta int
 	}
 
 	if workspaceRaw, hasWorkspaceId := d.GetOk("workspace_id"); hasWorkspaceId {
-		applicationInsightsComponentProperties.WorkspaceResourceID = utils.String(workspaceRaw.(string))
+		workspaceID, err := workspaces.ParseWorkspaceIDInsensitively(workspaceRaw.(string))
```

Comment:
since this is in the Create/Update function (and we're validating this above in the Schema) we can parse this case-sensitively here:

```suggestion
		workspaceID, err := workspaces.ParseWorkspaceID(workspaceRaw.(string))
```